File: StarBar/src/main/java/com/gcode/widget/StarBarSelectMethod.java
package com.gcode.widget;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author Vast Gui
 * @email sakurajimamai2020@qq.com
 */
public class StarBarSelectMethod {
    // Unable
    public static final int Unable = 0;
    // Click
    public static final int Click = 1;
    // Sliding
    public static final int Sliding = 2;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({
            Unable,Click,Sliding
    })
    public @interface SelectMethod { }
}

File: StarBar/src/main/java/com/gcode/widget/StarBarSelectMethod.kt
package com.gcode.widget

/**
 *作者:created by HP on 2021/8/1 17:17
 *邮箱:sakurajimamai2020@qq.com
 */
enum class StarBarSelectMethod(val value:Int) {
    //无法进行点击修改
    ProhibitedOperation(0),
    //点击选择星星数量
    ClickOperation(1),
    //滑动选择星星数量
    SlidingOperation(2)
}